src/components/CheckboxList.tsx

```diff
@@ -0,0 +1,57 @@
+import { blueGrey700 } from "@/constants";
+import { addOpacityToHsla } from "@/helpers";
+import type { Filter, FilterOnCheckedChange, Filters } from "@/types";
+import styled from "styled-components";
+import { Checkbox } from "@/components";
+
+interface Props {
+  filters: Filters;
+  onCheckedChange: FilterOnCheckedChange;
+}
+/**
+ * A list of checkboxes that is used in the filters component.
+ * @param filters The filters that are used to create the checkboxes.
+ * @param onCheckedChange A callback function that is called when a checkbox is checked or unchecked.
+ */
+export function CheckboxList({ filters, onCheckedChange }: Props) {
+  return (
+    <Wrapper>
+      {Object.entries(filters).map(([filter, items]) => (
+        <CheckboxesWrapper key={filter}>
+          <Title>{filter}</Title>
+          {Object.entries(items).map(([itemName, { checked, count }]) => (
+            <Checkbox
+              key={itemName}
+              checked={checked}
+              itemName={itemName}
+              count={count}
+              onCheckedChange={({ ...args }) =>
+                onCheckedChange({ ...args, filter: filter as Filter })
```

Comment:
surprised this needs to be cast

Reply:
yeah it's super annoying, happens because Object.entries loses the type of the values it iterates over. https://github.com/microsoft/TypeScript/issues/35101

Reply:
the alternative is to manually type the result of Object.entries which is even more annoying